packages/lastrev-build-prefetch-content/src/helpers/excludeContent.ts

```diff
@@ -0,0 +1,30 @@
+import jsonLogic from 'json-logic-js';
+import { Entry } from 'contentful';
+import { chain, get, identity, some } from 'lodash';
+import { ResolvedBuildConfig } from '../types';
+import getFieldValue from './getFieldValue';
+
+const excludeContent = (
+  buildConfig: ResolvedBuildConfig,
+  content: Entry<any>,
+  locale: string,
+  defaultLocale?: string
+): { excludeIfParentIsExcluded: boolean; exclude: boolean } => {
+  const contentTypeId = get(content, 'sys.contentType.sys.id');
+  const config = get(buildConfig, `excludePages.${contentTypeId}`);
+
+  if (!contentTypeId || !config) return { excludeIfParentIsExcluded: false, exclude: false };
+
+  const { rules, fields, excludeIfParentIsExcluded } = config;
+
+  const data = chain(fields)
+    .keyBy(identity)
+    .mapValues((fieldName) => getFieldValue(content, fieldName, locale, defaultLocale))
+    .value();
+
+  const exclude = !!jsonLogic.apply(rules, data);
```

Comment:
If `rules` or `data` is undefined/null will this fail?

Reply:
based on comment above, data will never be undefined. rules could be though, but json-logic also handles this without erroring. `!!` at the begining will ensure that that returns false.